Friendly Environment Policy

spdegabrielle opened the pull request:
Adapted from the [RacketCon 2018 Friendly Environment Policy](https://github.com/racket/racket-lang-org/blob/0d8a413ba37bfb1a0bebc7ea8ed5a9da4c76b55b/rcon/2018/index.html.pm#L278) and the [Rust Code of conduct](https://www.rust-lang.org/policies/code-of-conduct) which in turn was adapted from the Node.js Policy on Trolling as well as the Contributor Covenant v1.3.0.

First draft for discussion.

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+
+◊special-section[#:class "one-column-body-text" #:id "pull-quote"]{The Racket community aims to improve the world through programming. It started with the goal of introducing everyone to the wonderful world of program design, with a spirit of full inclusion and no emphasis on any specific group.  Over time it has grown into a full-fledged professional community with a well-known reputation for he ...
+
+This spirit requires an environment that enables all to participate without fear of personal harassment. We define harassment as unwelcome or hostile behavior, that is, behavior that focuses on people instead of ideas. ◊link["http://www.acm.org/sigs/volunteer_resources/officers_manual/anti-harassment-policy"]{The ACM's anti-harassment policy} lists a variety of specific unacceptable factors and b ...
```
Pls cut sentence with link too ACM.

spdegabrielle replied:
Ok. Will do

spdegabrielle replied:
done

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+◊section{
+Friendly Environment Policy
+
+* We are committed to providing a friendly, safe and welcoming environment for all, regardless of level of experience, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or other similar characteristic.
```
Pls replace "gender identity ... other similar characteristic" with "personal characteristic". 

This is in line with the first paragraph and includes everything, even if we don't think of it now (e.g. the belief that conservative politicians are people, too).

spdegabrielle replied:
Ok will do (away from keyboard so just reviewing comments - will action when I get back)

spdegabrielle replied:
removed 'gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or other similar'

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+* We are committed to providing a friendly, safe and welcoming environment for all, regardless of level of experience, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or other similar characteristic.
+* Please avoid using overtly sexual aliases or other nicknames that might detract from a friendly, safe and welcoming environment for all.
+* Please be kind and courteous. There’s no need to be mean or rude.
+* Respect that people have differences of opinion and that every design or implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
```
Cut "There is seldom a right answer." Sometimes there is (TR tries hard to live up to TS and that's not arguable after Sam and I agreed on the holes that exist.)

spdegabrielle replied:
Ok

spdegabrielle replied:
done

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+
+These are the policies for upholding our community’s standards of conduct. If you feel that a thread needs moderation, please contact the Rust moderation team .
+
+1. Remarks that violate the Racket standards of conduct, including hateful, hurtful, oppressive, or exclusionary remarks, are not allowed. (Cursing is allowed, but never targeting another user, and never in a hateful manner.)
```
"Cursing is allowed" .. I'd cut the parenthetical remark.

spdegabrielle replied:
Ok

spdegabrielle replied:
done

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+
+1. Remarks that violate the Racket standards of conduct, including hateful, hurtful, oppressive, or exclusionary remarks, are not allowed. (Cursing is allowed, but never targeting another user, and never in a hateful manner.)
+
+2. Remarks that moderators find inappropriate, whether listed in the code of conduct or not, are also not allowed.
```
This sounds whimsical and dictatorial.

spdegabrielle replied:
Would be good to remove the whimsy. I’d like no doubt that the moderators are in charge.

spdegabrielle replied:
What would you suggest?

mfelleisen replied:
How about "The moderators reserve the right to amend this policy."

spdegabrielle replied:
Done

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+
+5. If the user comes back and continues to make trouble, they will be banned, i.e., indefinitely excluded.
+
+6. Moderators may choose at their discretion to un-ban the user if it was a first offense and they offer the offended party a genuine apology.
```
"they" as in "dear firefight, they are upstairs in the burning house"? 

Pls use he/she.

spdegabrielle replied:
I prefer ‘the user’ in line with the rest of the document. Ok with you?

spdegabrielle replied:
I've just changed it to 'the user'.

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+
+In the Racket community we strive to go the extra step to look out for each other. Don’t just aim to be technically unimpeachable, try to be your best self. In particular, avoid flirting with offensive or sensitive issues, particularly if they’re off-topic; this all too often leads to unnecessary fights, hurt feelings, and damaged trust; worse, it can drive people away from the community entirely ...
+
+And if someone takes issue with something you said or did, resist the urge to be defensive. Just stop doing what it was they complained about and apologize. Even if you feel you were misinterpreted or unfairly accused, chances are good there was something you could’ve communicated better — remember that it’s your responsibility to make your fellow Racketeers comfortable. Everyone wants to get alo ...
```
This point is wholly inappropriate. It gives every single community member a "veto" on words and phrases and whatever. I found many words on "users" inappropriate when I participated and I just swallowed hard and let the technical point stand. (And when I could no longer stand it, I left.)

spdegabrielle replied:
You are right.  It is advice, and does not belong in policy. I've moved it to the preamble section.

> In the Racket community we strive to go the extra step to look out for each other. Don’t just aim to be technically unimpeachable, try to be your best self. In particular, avoid flirting with offensive or sensitive issues, particularly if they’re off-topic; this all too often leads to unnecessary fights, hurt feelings, and damaged trust; worse, it can drive people away from the community entirely.
> 
> If someone takes issue with something you said or did, resist the urge to be defensive. Just stop doing what it was they complained about and apologize. Even if you feel you were misinterpreted or unfairly accused, chances are good there was something you could’ve communicated better — remember that it’s your responsibility to make your fellow Racketeers comfortable. Everyone wants to get along and we are all here first and foremost because we want to talk about cool technology. You will find that people will be eager to assume good intent and forgive as long as you earn their trust.

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+
+And if someone takes issue with something you said or did, resist the urge to be defensive. Just stop doing what it was they complained about and apologize. Even if you feel you were misinterpreted or unfairly accused, chances are good there was something you could’ve communicated better — remember that it’s your responsibility to make your fellow Racketeers comfortable. Everyone wants to get alo ...
+
+The enforcement policies listed above apply to all official Racket venues; including the Racket Slack channel; 
```
List dev.

spdegabrielle replied:
I’ll add it explicitly.

spdegabrielle replied:
done

mfelleisen commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,70 @@
+The enforcement policies listed above apply to all official Racket venues; including the Racket Slack channel; 
+#racket IRC on freenode.net; GitHub repositories under github.com/racket; and all forums under lists.racket-lang.org (including https://groups.google.com/forum/#!forum/racket-users). 
+
+For other projects adopting the Rust Code of Conduct, please contact the maintainers of those projects for enforcement. If you wish to use this code of conduct for your own project, consider explicitly mentioning your moderation policy or making a copy with your own moderation policy so as to avoid confusion.
```
I hate Rust :-)

spdegabrielle replied:
I’ll remove the Rust & other references. Would love to know why but this is not the venue. You have my email if you care to share privately.

spdegabrielle replied:
Removed whole section.

jeapostrophe commented on www/friendly.html.pm:
```diff
@@ -0,0 +1,64 @@
+
+These are the policies for upholding our community’s standards of conduct. If you feel that a thread needs moderation, please contact the Rust moderation team .
+
+1. Remarks that violate the Racket standards of conduct, including hateful, hurtful, oppressive, or exclusionary remarks, are not allowed. Cursing is allowed, but never targeting another user, and never in a hateful manner.
```
I find this entire policy to be absurd and one-sided towards a very particular perspective of what is offensive and inappropriate behavior. For example, I believe that I am as offended by cursing and blasphemy as other people are by things they might call "hateful, hurtful, oppressive, or exclusionary." Yet, I don't think it is appropriate for me to impose my opinions about what words people should say on the Racket community. Being in the world but not of the world requires me to ignore people when they don't do what I would do and practice charitable forbearance with their behavior. I think the best thing for a Friendly-Environment would be 1 Corinthians 13 and D&C 88:133. I assume that many readers of this policy would find my suggestion absurd; if you do, that's how I feel about this document.

mfelleisen replied:
@jeapostrophe ~~ I have expressed a similar opinion here (with slightly softer words; see my comments on "user") and in email to @spdegabrielle But we live in times where such a policy is sadly desired by some people and even more sadly necessary to some extent (as the Scala community had to learn thru harsh experiences). 

As for the "cursing" I had actually marked it up and changed my mind. We should delete this sentence.

spdegabrielle replied:
@mfelleisen Sentence removed.

@jeapostrophe I'd like this document to be as universal as possible. Neither 1 Corinthians 13 or D&C 88:133 are absurd, though I had to look both up to my shame. The world needs more love and fellowship. Even our tiny corner. 

I'm hoping the preamble helps a little, but it could obviously be improved. 

> The Racket community aims to improve the world through programming. It started with the goal of introducing everyone to the wonderful world of program design, with a spirit of full inclusion and no emphasis on any specific group.  Over time it has grown into a full-fledged professional community with a well-known reputation for helpfulness and openness on its on-line communication channels. 
> 
> This spirit requires an environment that enables all to participate without fear of personal harassment. We define harassment as unwelcome or hostile behavior, that is, behavior that focuses on people instead of ideas. The organizers consider responses such as “just joking,” or “teasing,” or being “playful” as unacceptable.
> 
> In the Racket community we strive to go the extra step to look out for each other. Don’t just aim to be technically unimpeachable, try to be your best self. In particular, avoid flirting with offensive or sensitive issues, particularly if they’re off-topic; this all too often leads to unnecessary fights, hurt feelings, and damaged trust; worse, it can drive people away from the community entirely.
> 
> If someone takes issue with something you said or did, resist the urge to be defensive. Just stop doing what it was they complained about and apologize. Even if you feel you were misinterpreted or unfairly accused, chances are good there was something you could’ve communicated better — remember that it’s your responsibility to make your fellow Racketeers comfortable. Everyone wants to get along and we are all here first and foremost because we want to talk about cool technology. You will find that people will be eager to assume good intent and forgive as long as you earn their trust.

lexi-lambda replied:
The purpose of a code of conduct is to communicate the intent of the moderation team—it is descriptive, not prescriptive. If the words in a code of conduct do not align with what will actually be enforced, then it is a bad code of conduct, and it is arguably worse than having no code of conduct at all. Therefore, the moderation team should write the code of conduct, if one is provided at all.

Obviously, the presence or absence of a code of conduct (and its contents) can influence which people you attract to your community (whether you like it or not). But that is because people are willing to trust, on some level, that what is contained within the code of conduct reflects the values of the community. If that isn’t true, a code of conduct serves no purpose.

spdegabrielle replied:
thank you.

mfelleisen commented:
> On Sep 1, 2019, at 4:10 PM, Stephen De Gabrielle <notifications@github.com> wrote:
> 
> The Racket community aims to improve the world through programming. It started with the goal of introducing everyone to the wonderful world of program design, with a spirit of full inclusion and no emphasis on any specific group. Over time it has grown into a full-fledged professional community with a well-known reputation for helpfulness and openness on its on-line communication channels.
> 
> This spirit requires an environment that enables all to participate without fear of personal harassment. We define harassment as unwelcome or hostile behavior, that is, behavior that focuses on people instead of ideas. The organizers consider responses such as “just joking,” or “teasing,” or being “playful” as unacceptable.
> 
> In the Racket community we strive to go the extra step to look out for each other. Don’t just aim to be technically unimpeachable, try to be your best self. In particular, avoid flirting with offensive or sensitive issues, particularly if they’re off-topic; this all too often leads to unnecessary fights, hurt feelings, and damaged trust; worse, it can drive people away from the community entirely.
> 
> If someone takes issue with something you said or did, resist the urge to be defensive. Just stop doing what it was they complained about and apologize. Even if you feel you were misinterpreted or unfairly accused, chances are good there was something you could’ve communicated better — remember that it’s your responsibility to make your fellow Racketeers comfortable. Everyone wants to get along and we are all here first and foremost because we want to talk about cool technology. You will find that people will be eager to assume good intent and forgive as long as you earn their trust.
> 
> 


I still find the last paragraph offensive and would love to see it removed. It is absolutely not anyone’s responsibility to make another person happy” because it is impossible. Every psychologist will tell you tat.  (Believe me, been there tried that.) Now exchange “happy” with “comfortable” and you see the problem. Someone can easily pretend to be uncomfortable and thus shut down any content. 

I am even more offended by “apologize”. Science has to be aggressive to find the truth. Your best friends are those who aggressively try to poke holes into your work. I think extensions of science, such as PL design, and derivative science work (such as program design) can only benefit from such frank exchanges. And if I have to apologize for having stepped on someone’s toes or if someone needs to apologize for having stepped on my toes, such frank dialogue is inhibited.

spdegabrielle commented:
@mfelleisen last paragraph removed

>If someone takes issue with something you said or did, resist the urge to be defensive. Just stop doing what it was they complained about and apologize. Even if you feel you were misinterpreted or unfairly accused, chances are good there was something you could’ve communicated better — remember that it’s your responsibility to make your fellow Racketeers comfortable. Everyone wants to get along and we are all here first and foremost because we want to talk about cool technology. You will find that people will be eager to assume good intent and forgive as long as you earn their trust.

mfelleisen commented:
> On Sep 1, 2019, at 4:24 PM, Stephen De Gabrielle <notifications@github.com> wrote:
> 
> @mfelleisen <https://github.com/mfelleisen> last paragraph removed
> 
> 


Thank you.

spdegabrielle commented:
added; 
> Derived from RacketCon 2018.

samth commented:
I feel like this is premature in two senses:

1. Adopting a code of conduct for the Racket project is something that should be discussed before implementing, and there hasn't been such a discussion yet. Starting with a particular text is probably not the best way to do that. 

2. The Rust Code of Conduct, while good in many ways, assumes the existence of a set of moderators that our project doesn't have, and that would take volunteers and discussion to choose, if we even want to have such a group.

mfelleisen commented:
1. In the spirit of the "hack" rule, I think that a discussion is productive when it starts from a draft. 

2. As Matthew says about Racket2 "it can fail" and then we conclude not to have a code.

3. Since you brought up the Rust code again, I will say that it is ironize for a company whose intolerant employees (and others) drove out its CEO/CTO for his political opinion like a lynch mob of the olden days. ~~ In this spirit, I emphasize 2. 

So now that we have a draft document I think we can use this RFC to discuss whether we want one, whether additional fixes to Stephen's draft will yield an acceptable one, or whether a different approach is warranted. 

@spdegabrielle Again thank you for taking the initiative and please don't be discouraged if we decided to go a different route.

spdegabrielle commented:
I'm very encouraged this has had so much engagement, and I thank you all for this. 

@mfelleisen and @jeapostrophe thank you both for you feedback I believe we have already made some progress from the initial draft.

Issues raised above (please correct me if I have misrepresented)
1. adoption of CoC requires discussion and agreement before implementation - @samth 
2. this is a draft and as such if can be changed or rejected completely - @mfelleisen 
3. the CoC must not discourage or close down debate - @mfelleisen 
4. the CoC must reflect the values of the community - @lexi-lambda 
5. the CoC must match what is enforced by the moderation team - @lexi-lambda  
6. a moderation team is required for enforcement - @samth 

For reference I've summarised earlier issues and changes in original draft (see above)
1. removed link to ACM Anti-harassment policy
2. simplified the statement of intent to ' safe and welcoming environment for all' (removed incomplete list of categories - but left in 'level of experience')
3. Cut "There is seldom a right answer."
4. replaced "Remarks that moderators find inappropriate, whether listed in the code of conduct or not..." with "The moderators reserve the right to amend this policy."
5. cut "Cursing is allowed"
6. Switched to non-gendered terminology. "the user"
7. included racket-dev

spdegabrielle commented:
I missed an item 

7. **the moderation team should write the code of conduct**, if one is provided at all. (thanks @lexi-lambda )

Rather than the core team taking on this task themselves, I believe they should appoint the initial membership. They could appoint themselves, call for volunteers, or directly approach community members who they trust with the task, with the understanding the core team can change the membership of the moderation team at any time, and 
a) the first and only task of the moderation team is to write the code of conduct, 
b) only the core team has the authority to approve the CoC

I believe this is a lightweight approach that would be appropriate for a community of this size. 

Kind regards, 

Stephen

PS I believe the core team consists of; @mfelleisen @mflatt @rfindler @jeapostrophe @samth is that right?

mfelleisen commented:
If CoC captors the values of the community, I don't think it is necessary for the moderators to write it. Presumably Racket N will live forever and people don't, so sooner or later the crew of moderators will have to change. 

I have pointed the core team to your proposal. Let's see what the discussion says. 

Again, thanks for taking the initiative.

spdegabrielle commented:
> If CoC captors the values of the community, I don't think it is necessary for the moderators to write it. 

Making the moderation team write it serves two purposes; engaging the community and taking the load off the core team. As the core team decide the composition of the moderation team, while retaining veto and membership control, I'd hope this was a low workload/low risk approach

Presumably Racket N will live forever and people don't, so sooner or later the crew of moderators will have to change.
> 
> I have pointed the core team to your proposal. Let's see what the discussion says.

thank you 

> 
> Again, thanks for taking the initiative.

no problem

s.

spdegabrielle commented:
@mflatt closing

mflatt commented:
Thanks, @spdegabrielle !

Closing this PR doesn't mean that there won't be a Friendly Environment Policy on the web page – only that we currently think that starting with something closer to the RacketCon 2018 text will be better. I'll reference this PR from the new one to carry over discussion.

mflatt commented:
Continued as #112
